### PR TITLE
Provide update resource functionality.

### PR DESCRIPTION
**Pull request by osahyoun**

### CODE REVIEW

File: lib/ak/updater.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Monkey Patch!
+# To be moved and spec'd at https://github.com/EricBoersma/actionkit_connector
+module ActionKitConnector
```
Comment (osahyoun) [inline lib/ak/updater.rb]: Will move when working on staging.

File: app/workers/queue_listener.rb [hunk 2]
```diff
@@ -28,6 +30,10 @@ def create_action(params)
     AkActionCreator.create_action(params[:params][:slug], params[:params][:body])
   end
 
+  def update_resource(params)
+    Ak::Updater.update(uri: params[:uri], body: params[:params])
```
Comment (NealJMD) [inline app/workers/queue_listener.rb]: It would be nice if the naming scheme between `Ak::Updater` and `AkPageCreator` was consistent

File: lib/ak/updater.rb (hunk 1 above)
Comment (NealJMD) [inline lib/ak/updater.rb]: I'm sorry I don't understand, could you clarify what's going on?

File: lib/ak_page_creator.rb [hunk 3]
```diff
@@ -8,36 +8,45 @@ class << self
     def create_page(name, title, language, url, type, page_id)
       new.create_page(name, title, language, url, type, page_id)
     end
+
+    def page_types
```
Comment (NealJMD) [inline lib/ak_page_creator.rb]: does this get called?

File: lib/ak_page_creator.rb [hunk 4]
```diff
@@ -8,36 +8,45 @@ class << self
...
     page = Page.find(page_id)
 
-    if request.response.class == Net::HTTPCreated
-      page.status = 'success'
-      page.save
-    elsif request.response.class == Net::HTTPBadRequest
-      page.status = 'failed'
-      page.messages = request.parsed_response.to_json
-      page.save
+    case request.response
```
Comment (NealJMD) [inline lib/ak_page_creator.rb]: why doesn't this have to be `case request.response.class`?

File: lib/ak_page_creator.rb (hunk 3 above)
Comment (NealJMD) [inline lib/ak_page_creator.rb]: oh I see you just moved it

File: spec/requests/message_handler_spec.rb [hunk 5]
```diff
@@ -1,7 +1,30 @@
...
+        params: {
+          title: 'Foo Bar',
+          tags: ["/rest/v1/tag/878/"]
+        }
+      }
+    end
+
+    context "204 response" do
+      it "updates resource" do
+        VCR.use_cassette('update_resource_204') do
```
Comment (NealJMD) [inline spec/requests/message_handler_spec.rb]: what are you going for with `use_cassette`? I know that it's not necessary - you can just do 
```
it "updates resource" ,:vcr do
 ...
end
```
and it will create a cassette just for that test. but is there a different functionality you're going for? I don't see you reusing these casettes...

File: lib/ak/updater.rb (hunk 1 above)
Comment (osahyoun) [inline lib/ak/updater.rb]: This code is adding to https://github.com/EricBoersma/actionkit_connector - I want to have a working build on staging before updating Eric's gem and having a new version cut. Does that make sense?

File: app/workers/queue_listener.rb (hunk 2 above)
Comment (osahyoun) [inline app/workers/queue_listener.rb]: I want to namespace the whole API with `AK`, and this is my first step. The rest will be refactored in another PR.

File: spec/requests/message_handler_spec.rb (hunk 5 above)
Comment (osahyoun) [inline spec/requests/message_handler_spec.rb]: `VCR.user_cassette` is the API I know. Sounds like it does the same thing at `:vcr`

File: lib/ak_page_creator.rb (hunk 4 above)
Comment (osahyoun) [inline lib/ak_page_creator.rb]: `case` uses `===`.

```ruby

class Foo
end

a = Foo.new

case a
when Foo
  puts 'FOOOO'
else
  puts 'no'
end

Foo === Foo.new

# FOOOO
# true
```

Comment (NealJMD) [inline lib/ak_page_creator.rb]: ughh that's so confusing! after reading about `===` I don't want to use statements any more. It's so uninuitive that case doesn't do equality matching!

File: spec/requests/message_handler_spec.rb (hunk 5 above)
Comment (NealJMD) [inline spec/requests/message_handler_spec.rb]: conform everything to my style!!!!!!

Comment (NealJMD) [inline spec/requests/message_handler_spec.rb]: jk iz fine

**Comment by osahyoun** [comment]
Thanks, mate.
